cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -420,13 +459,24 @@ private LocalFile uploadCacheableFile(String bucket, String targetFilePath,
    * Uploads files to gcs.
    */
   private LocalFile uploadFile(String bucket, String targetFilePath,
-                               LocalFile localFile, boolean isCacheable)
+                               LocalFile localFile, boolean cacheable)
     throws IOException, StorageException {
     BlobId blobId = BlobId.of(bucket, targetFilePath);
     String contentType = "application/octet-stream";
     BlobInfo.Builder blobInfoBuilder = BlobInfo.newBuilder(blobId);
-    if (isCacheable) {
-      blobInfoBuilder.setCustomTime(System.currentTimeMillis());
+    // don't set custom time on artifacts cacheable per CDAP version.
+    if (cacheable) {
+      if (artifactsCacheablePerCDAPVersion.contains(localFile.getName())) {
+        if (cdapVersionInfo.isSnapshot()) {
+          // if the version is SNAPSHOT, set a custom time of buildTime + 7 days for cleanup.
+          blobInfoBuilder.setCustomTime(cdapVersionInfo.getBuildTime() + TimeUnit.DAYS.toMillis(7));
+        } else {
+          // if the version is non-snapshot, set a custom time of buildTime + 2years for cleanup.
+          blobInfoBuilder.setCustomTime(cdapVersionInfo.getBuildTime() + TimeUnit.DAYS.toMillis(730));
+        }
+      } else {
+        blobInfoBuilder.setCustomTime(System.currentTimeMillis());
```

Comment:
It's better to use the `if` condition to compute the customTime and just have one place to call `setCustomTime`. The logic would be easier to follow.

```
if (cacheable) {
  long customTime = System.currentTimeMillis();
  if (artifactsCacheablePerCDAPVersion.contains(localFile.getName())) {
    customTime = cdapVersionInfo.getBuildTime() 
      + TimeUnit.DAYS.toMillis(cdapVersionInfo.isSnapshot() ? SNAPSHOT_EXPIRE_DAYS : EXPIRE_DAYS);
  }
  buildInfoBuilder.setCustomTime(customTime);
}
```
Also, maybe make `7` and `730` as some short private constants

Reply:
ack